From d197d4b389daabbad20468ea59f62f94402d7e89 Mon Sep 17 00:00:00 2001
From: Don Zickus <dzickus@redhat.com>
Date: Fri, 12 Sep 2014 18:13:01 -0400
Subject: [powerpc] Exported functions __clear_user and copy_page use r2 so need _GLOBAL_TOC()

Message-id: <1410545655-205645-553-git-send-email-dzickus@redhat.com>
Patchwork-id: 94662
O-Subject: [RHEL7 PATCH 552/626] powerpc: Exported functions __clear_user and copy_page use r2 so need _GLOBAL_TOC()
Bugzilla: 1127366
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Steve Best <sbest@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>
RH-Acked-by: David Gibson <dgibson@redhat.com>

https://bugzilla.redhat.com/show_bug.cgi?id=1127366

commit 2ac7b0166a9bc743893be3f38e7b1729ce975ab9
Author: Anton Blanchard <anton@samba.org>
Date:   Thu Jun 5 08:04:39 2014 +1000

    powerpc: Exported functions __clear_user and copy_page use r2 so need _GLOBAL_TOC()

    __clear_user and copy_page load from the TOC and are also exported
    to modules. This means we have to use _GLOBAL_TOC() so that we
    create the global entry point that sets up the TOC.

    Signed-off-by: Anton Blanchard <anton@samba.org>
    Signed-off-by: Benjamin Herrenschmidt <benh@kernel.crashing.org>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/arch/powerpc/lib/copypage_64.S b/arch/powerpc/lib/copypage_64.S
index e59c9c2..a3c4dc4 100644
--- a/arch/powerpc/lib/copypage_64.S
+++ b/arch/powerpc/lib/copypage_64.S
@@ -16,7 +16,7 @@ PPC64_CACHES:
         .tc             ppc64_caches[TC],ppc64_caches
         .section        ".text"
 
-_GLOBAL(copy_page)
+_GLOBAL_TOC(copy_page)
 BEGIN_FTR_SECTION
  lis r5,PAGE_SIZE@h
 FTR_SECTION_ELSE
diff --git a/arch/powerpc/lib/string_64.S b/arch/powerpc/lib/string_64.S
index 3b1e480..7bd9549 100644
--- a/arch/powerpc/lib/string_64.S
+++ b/arch/powerpc/lib/string_64.S
@@ -77,7 +77,7 @@ err3; stb r0,0(r3)
  mr r3,r4
  blr
 
-_GLOBAL(__clear_user)
+_GLOBAL_TOC(__clear_user)
  cmpdi r4,32
  neg r6,r3
  li r0,0
-- 
1.7.1